src/column-layout/internal.tsx

```diff
@@ -36,31 +29,31 @@ export default React.forwardRef(function ColumnLayout(
   ref?: React.Ref<any>
 ) {
   const baseProps = getBaseProps(restProps);
-  const isTextGridVariant = variant === 'text-grid';
-  const shouldDisableGutters = !isTextGridVariant && disableGutters;
-  const shouldHaveHorizontalBorders = !isTextGridVariant && (borders === 'horizontal' || borders === 'all');
-  const shouldHaveVerticalBorders = !isTextGridVariant && (borders === 'vertical' || borders === 'all');
-  /*
-   Flattening the children allows us to "see through" React Fragments and nested arrays.
-   */
-  const flattenedChildren = flattenChildren(children);
 
   return (
     <div {...baseProps} className={clsx(baseProps.className, styles['column-layout'])} ref={__internalRootRef}>
-      <InternalGrid
-        ref={ref}
-        disableGutters={true}
-        gridDefinition={repeat(COLUMN_DEFS[columns] ?? {}, flattenedChildren.length)}
-        className={clsx(styles.grid, styles[`grid-columns-${columns}`], styles[`grid-variant-${variant}`], {
-          [styles['grid-horizontal-borders']]: shouldHaveHorizontalBorders,
-          [styles['grid-vertical-borders']]: shouldHaveVerticalBorders,
-          [styles['grid-no-gutters']]: shouldDisableGutters,
-        })}
-        __breakpoint={__breakpoint}
-        __responsiveClassName={breakpoint => breakpoint && styles[`grid-breakpoint-${breakpoint}`]}
-      >
-        {children}
-      </InternalGrid>
+      {minColumnWidth ? (
+        <ColumnLayoutWithCSS
+          columns={columns}
+          borders={borders}
+          variant={variant}
+          minColumnWidth={minColumnWidth}
+          disableGutters={disableGutters}
+        >
+          {children}
+        </ColumnLayoutWithCSS>
+      ) : (
+        <ColumnLayoutWithGrid
+          columns={columns}
+          variant={variant}
+          borders={borders}
+          disableGutters={disableGutters}
+          __breakpoint={__breakpoint}
+          ref={ref}
```

Comment:
Why do we forward ref to one instance, but not another? If it is some deprecated usage we want to avoid in the new code, it needs a comment, to prevent future developers "fixing" this bug

Reply:
I was wondering this too and I couldn't figure out what this ref is for. So yeah, I'm happy to add a comment.

Reply:
This is needed only for this line: https://github.com/cloudscape-design/components/blob/d6b4e10cdd6684bbe9c86149d649902bca3b73af/src/column-layout/index.tsx#L21

Can be simplified by moving it down to "with grid" component